distributed/shuffle/_core.py

```diff
@@ -584,6 +584,7 @@ def p2p_barrier(id: ShuffleId, *run_ids: int) -> int:
 
 class P2PBarrierTask(Task):
     spec: ShuffleSpec
+    block_fusion: bool
```

Comment:
This would require `hasattr` in the fusion code, wouldn't it? I like the idea with the property on the baseclass better

Reply:
No, I defined it over in dask/dask on every class (but that's ugly in a different way)...